src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -233,7 +197,8 @@ public async Task InitializeAsync(CancellationToken cancellationToken)
                     // That's because AddComponentAsync waits for quiescence, which can take
                     // arbitrarily long. In the meantime we might need to be receiving and
                     // processing incoming JSInterop calls or similar.
-                    for (var i = 0; i < Descriptors.Count; i++)
+                    var count = Descriptors.Count;
+                    for (var i = 0; i < count; i++)
                     {
                         var (componentType, domElementSelector, prerendered) = Descriptors[i];
                         if (!prerendered)
```

Comment:
Is this flag still required?

Reply:
Also why does `ComponentDescriptor` need deconstruction?

Reply:
Unrelated, could you format this document? The formatting is off on Line 224

Reply:
huh, wierd I thought I nuked that. will nuke completely.

Reply:
done, and formatted.